src/com/goide/GoModuleBuilder.java

```diff
@@ -58,5 +60,8 @@ public boolean isSuitableSdkType(SdkTypeId sdkType) {
   @Override
   public void moduleCreated(@NotNull Module module) {
     CompilerWorkspaceConfiguration.getInstance(module.getProject()).CLEAR_OUTPUT_DIRECTORY = false;
+    ModifiableRootModel rootModel = ModuleRootManager.getInstance(module).getModifiableModel();
+    rootModel.inheritSdk();
+    ApplicationManager.getApplication().runWriteAction(rootModel::commit);
```

Comment:
Changing model of module while handling listener's event changing model of modules looks very suspisious I'd rather avoid this.
Setting inheritSdk happens in com.intellij.ide.util.projectWizard.JavaModuleBuilder#setupRootModel, it's better to fingure out why it doesn't work as expected.